### PR TITLE
[Merged by Bors] - feat(Pullback/CommSq): add more pasting lemmas for `IsPullback`

### PR DESCRIPTION
**Pull request by callesonne**
This PR adds two (variants) of pasting lemmas in the `IsPullback` framework. These are variants where the second square has a morphism that is induced from the universal property of the first.

This work was inspired by the AIM workshop "Formalizing algebraic geometry" in June 2024.

---

These lemmas are used in PR #14208

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean [hunk 1]
```diff
@@ -201,6 +201,21 @@ theorem cone_snd (h : IsPullback fst snd f g) : h.cone.snd = snd :=
...
+@[reassoc (attr := simp)]
+lemma lift_fst (hP : IsPullback fst snd f g) {W : C} (h : W ⟶ X) (k : W ⟶ Y)
+    (w : h ≫ f = k ≫ g) : hP.lift h k w ≫ fst = h :=
+  PullbackCone.IsLimit.lift_fst hP.isLimit h k w
+
+@[reassoc (attr := simp)]
+lemma lift_snd (hP : IsPullback fst snd f g) {W : C} (h : W ⟶ X) (k : W ⟶ Y)
+    (w : h ≫ f = k ≫ g) : hP.lift h k w ≫ snd = k :=
+  PullbackCone.IsLimit.lift_snd hP.isLimit h k w
+
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean:218]: Here, you may also add a `hom_ext` lemma: a morphism to `P` is determined by the two compositions. (Same for pushouts.)

File: Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean [hunk 2]
```diff
@@ -249,6 +264,38 @@ theorem of_hasBinaryProduct [HasBinaryProduct X Y] [HasZeroObject C] [HasZeroMor
...
+variable {P': C} {fst' : P' ⟶ X} {snd' : P' ⟶ Y}
+
+/-- Any object at the top left of a pullback square is isomorphic to the object at the top left
+of any other pullback square with the same cospan. -/
+noncomputable def isoIsPullback (h : IsPullback fst snd f g) (h' : IsPullback fst' snd' f g) :
+    P ≅ P' :=
+  IsLimit.conePointUniqueUpToIso h.isLimit h'.isLimit
+
+@[simp]
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean:277]: This also applies to other lemmas (basically anytime the LHS is a composition):
```suggestion
@[reassoc (attr := simp)]
```

Comment (callesonne) [reply Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean]: Thanks, forgot about that!

File: Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean (hunk 1 above)
Comment (callesonne) [reply Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean:218]: Fixed :)

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**